### PR TITLE
cp: fix compiler warnings

### LINKED ISSUE #1077: Compiler warnings for cp.rs
For version 1.20 (current stable) rustc gives several warnings on building cp.rs:

```
warning: unused imports: `c_char`, `c_int`
  --> src/cp/cp.rs:54:12
   |
54 | use libc::{c_int, c_char};
   |            ^^^^^  ^^^^^^
   |
   = note: #[warn(unused_imports)] on by default
warning: value assigned to `inode` is never read
   --> src/cp/cp.rs:684:17
    |
684 |             let mut inode: u64 = 0;
    |                 ^^^^^^^^^
    |
    = note: #[warn(unused_assignments)] on by default
warning: value assigned to `nlinks` is never read
   --> src/cp/cp.rs:685:17
    |
685 |             let mut nlinks = 0;
    |                 ^^^^^^^^^^
warning: unused variable: `src_path`
   --> src/cp/cp.rs:753:25
    |
753 |                     let src_path = CString::new(Path::new(&source.clone()).as_os_str().to_str().unwrap()).unwrap();
    |                         ^^^^^^^^
    |
    = note: #[warn(unused_variables)] on by default
warning: unused `std::result::Result` which must be used
   --> src/cp/cp.rs:910:25
    |
910 |                         xattr::set(dest, attr, &attr_value[..]);
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: #[warn(unused_must_use)] on by default
warning: variable `preserve_context` is assigned to, but never used
   --> src/cp/cp.rs:991:9
    |
991 |     let mut preserve_context = false;
    |         ^^^^^^^^^^^^^^^^^^^^
warning: value assigned to `preserve_context` is never read
   --> src/cp/cp.rs:994:13
    |
994 |             preserve_context = true;
    |             ^^^^^^^^^^^^^^^^
```

Taken from travis build starting here: https://travis-ci.org/uutils/coreutils/jobs/283112688#L617

I'll provide a PR :slightly_smiling_face:

### PR DESCRIPTION
**Pull request by shutefan**
Fixes #1077

### CODE REVIEW

**Comment by Arcterus** [comment]
Can you change all the doc comments the nightly is complaining about into regular comments (or just disable the warning)?

**Comment by shutefan** [comment]
I'll give it a try!

**Comment by shutefan** [comment]
I fixed some more `mut` related warnings but I'm not sure if trying to fix the rustdoc ones makes sense - apparently it is an error that is supposed to be fixed soon: https://github.com/rust-lang/rust/issues/43974

File: src/cp/cp.rs [hunk 1]
```diff
@@ -988,13 +984,6 @@ fn copy_file(source: &Path, dest: &Path, options: &Options) -> CopyResult<()> {
         println!("{}", context_for(source, dest));
     }
 
-    let mut preserve_context = false;
```
Comment (Arcterus) [inline src/cp/cp.rs]: Do you mind commenting this out or something and mention that it is for the `context` argument (which hasn't been implemented yet)?

**Comment by Arcterus** [comment]
Ah, I was not aware that it was a bug in rustdoc.  No need to fix the comments then.

File: src/cp/cp.rs (hunk 1 above)
Comment (shutefan) [reply src/cp/cp.rs]: Done. Had to wrap it in a block for the allow macro to cover the reassignment.

**Comment by Arcterus** [comment]
Thanks :)

**Pull request merged**